packages/next/src/server/base-server.ts

```diff
@@ -2455,7 +2464,10 @@ export default abstract class Server<
         typeof postponed === 'string' ||
         // If this is a dynamic RSC request, then this render supports dynamic
         // HTML (it's dynamic).
-        isDynamicRSCRequest
+        isDynamicRSCRequest ||
+        // When we're not sending the postponed state, we don't serve the partial shell as well.
+        // Instead we need to proceed a full dynamic rendering.
+        skipPostponed
```

Comment:
The `supportsDynamicResponse` flag is used to turn on/off blocking the Response stream on the entire render finishing. For `html` bots we need to no only do a full render we also need to not start reading the response until the entire render is complete. This is non-streaming mode basically. The reason is we need the metadata to be in the head when we flush. So I think you want to actually change this to `... || !skipPostponed`.

Reply:
I need to clean up the conditions after streaming metadata gets stable.

We have 2 things:
- supportsDynamicResponse - wether we can serve streaming
- isHtmlBotRequest - we should just return the full and blocking content 

But in worker store, `isStaticGeneration = !supportsDynamicResponse`
In PPR world, it will prerender, but not produce the full content.
That one condition sounds not enough for PPR mode of blocking rendering.

I leveraged the existing flags but later when streaming metadata gets stable, we should refactor the conditions